R/data-storage-log-file.R

```diff
@@ -0,0 +1,185 @@
+    },
+
+    #' @description read all user data from SQLite
+    #' @param date_from date representing the starting day of results
+    #' @param date_to date representing the last day of results
+
+    read_user_data = function(date_from, date_to) {
+      log_data <- private$read_data(private$log_file_path, date_from, date_to)
+
+      if (NROW(log_data) > 0) {
+        return(dplyr::mutate(log_data, date = as.Date(.data$time)))
+      }
+      log_data
+    },
+
+    #' @description read all session data from SQLite
+    #' @param date_from date representing the starting day of results
+    #' @param date_to date representing the last day of results
+
+    read_session_data = function(date_from, date_to) {
+      db_data <- private$read_data(private$session_file_path, date_from, date_to)
+
+      db_data %>%
+        dplyr::select("session", "detail") %>%
+        dplyr::group_by(.data$session) %>%
+        dplyr::summarise(title = paste(.data$detail, collapse = " | "))
+    },
+
+    #' @description read all session data
+
+    close = function() {
+      private$close_connection()
+    }
+  ),
+  active = list(
+
+    #' @field action_bucket string that identifies the file path to store user
+    #' related and action data
+
+    action_bucket = function() {
+      private$log_file_path
+    },
+
+    #' @field session_bucket string that identifies the bucket to store session
+    #' details data
+
+    session_bucket = function() {
+      private$session_file_path
+    }
+  ),
+  #
+  # Private
+  private = list(
+    # Private fields
+    log_file_path = NULL,
+
+    session_file_path = NULL,
+
+    # Private methods
+
+    # @name connect
+    # Makes connection to database based on passed config data
+    # @param log_file_path string with path to file for user actions
+    # @param session_file_path string with path to file for session details
+
+    connect = function(log_file_path, session_file_path) {
+      private$log_file_path <- log_file_path
+      private$session_file_path <- session_file_path
+    },
+
+    # @description reverts logger settings to default
+
+    close_connection = function() {
+    },
+
+    insert_checks = function(values, add_username) {
+      if ("time" %in% names(values)) {
+        rlang::abort(paste0(
+          "You must not pass 'time' value into database.",
+          " It is set automatically."
+        ))
+      }
+
+      if ("session" %in% names(values)) {
+        rlang::abort(paste0(
+          "You must not pass 'session' value into database.",
+          " It is set automatically."
+        ))
+      }
+
+      if ("username" %in% names(values)) {
+        rlang::abort(paste0(
+          "You must not pass 'username' value into database.",
+          " It is set automatically."
+        ))
+      }
+
+      values$time <- Sys.time()
+      values$session <- private$.session_id
+
+      if (add_username) {
+        values$username <- private$.username
+      }
```

Comment:
I initially thought this was not necessary until I saw how `log_custom_action()` was implemented.

Reply:
plumber adds a new layer of complexity with this (values need to be forced). We need to talk about how to best design/implement this interface.